app/decorators/submission_trait_scores_upload_decorator.rb

```diff
@@ -12,14 +12,47 @@ def parser_summary
           group_by{ |id, scores| scores.size }.
           sort{ |score1, score2| score2.first <=> score1.first }
 
-        summary << " - parsed #{trait_scores.size} plant(s) with unique identification"
+        summary << " - parsed #{trait_scores.size} plant scoring unit(s) with unique identification"
         histogram.each do |scoring_number, plants|
-          summary << "  - #{plants.size} plant(s) have #{scoring_number} trait score(s) recorded"
+          summary << "  - #{plants.size} unit(s) have #{scoring_number} trait score(s) recorded"
         end
       end
 
+      accessions = object.submission.content.step04.accessions
```

Comment:
I tried to "flatten"  `Submission#content` structure as an experiment, so that instead of `content.step04.trait_scores` we would write `content.trait_scores`, but I am not sure it is worth it. What do you think? The downside (or maybe not?) is that we would need to be more explicit about data we use to create some objects in finalizer. Eg. right now we use all data from the first step for PT `attrs = submission.content.step01.to_h.merge(common_data)`.